src/main/java/swift/ui/MainWindow.java

```diff
@@ -180,40 +181,45 @@ private void handleExit() {
      */
     @FXML
     private void handleTaskTab() {
-        logic.showAllLists();
-
-        personListPanel.removeHeading();
-        taskListPanel = new TaskListPanel(logic.getFilteredTaskList(), logic.getFilteredBridgeList(),
+        taskListPanel = new TaskListPanel(logic.getFilteredTaskList(), logic.getUnfilteredBridgeList(),
                 logic.getFilteredPersonList());
+
+        listPanelPlaceholder.getChildren().clear();
         listPanelPlaceholder.getChildren().add(taskListPanel.getRoot());
 
-        personTaskListPanel.removeHeading();
-        taskPersonListPanel = new TaskPersonListPanel(logic.getFilteredPersonList());
-        personTaskListPanelPlaceholder.getChildren().add(taskPersonListPanel.getRoot());
+        if (logic.getFilteredTaskList().size() == 1) {
+            taskPersonListPanel = new TaskPersonListPanel(logic.getFilteredPersonList());
+            personTaskListPanelPlaceholder.getChildren().clear();
+            personTaskListPanelPlaceholder.getChildren().add(taskPersonListPanel.getRoot());
+        } else {
+            personTaskListPanelPlaceholder.getChildren().clear();
+        }
     }
 
     /**
      * Switch to contacts tab.
      */
     @FXML
     private void handleContactTab() {
```

Comment:
```suggestion
    private void displayContactTab() {
```
`showContactTab()` or `displayContactTab()` up to you